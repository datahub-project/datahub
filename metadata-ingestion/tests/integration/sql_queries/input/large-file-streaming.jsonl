{"query": "SELECT * FROM snowflake.db.users WHERE user_id = 1", "timestamp": 1609459200, "user": "user_1", "operation_type": "SELECT"}
{"query": "SELECT * FROM snowflake.db.users WHERE user_id = 2", "timestamp": 1609459201, "user": "user_2", "operation_type": "SELECT"}
{"query": "SELECT * FROM snowflake.db.users WHERE user_id = 3", "timestamp": 1609459202, "user": "user_3", "operation_type": "SELECT"}
{"query": "SELECT * FROM snowflake.db.users WHERE user_id = 4", "timestamp": 1609459203, "user": "user_4", "operation_type": "SELECT"}
{"query": "SELECT * FROM snowflake.db.users WHERE user_id = 5", "timestamp": 1609459204, "user": "user_5", "operation_type": "SELECT"}
{"query": "SELECT * FROM snowflake.db.users WHERE user_id = 6", "timestamp": 1609459205, "user": "user_6", "operation_type": "SELECT"}
{"query": "SELECT * FROM snowflake.db.users WHERE user_id = 7", "timestamp": 1609459206, "user": "user_7", "operation_type": "SELECT"}
{"query": "SELECT * FROM snowflake.db.users WHERE user_id = 8", "timestamp": 1609459207, "user": "user_8", "operation_type": "SELECT"}
{"query": "SELECT * FROM snowflake.db.users WHERE user_id = 9", "timestamp": 1609459208, "user": "user_9", "operation_type": "SELECT"}
{"query": "SELECT * FROM snowflake.db.users WHERE user_id = 10", "timestamp": 1609459209, "user": "user_10", "operation_type": "SELECT"}
{"query": "SELECT * FROM snowflake.db.users WHERE user_id = 11", "timestamp": 1609459210, "user": "user_11", "operation_type": "SELECT"}
{"query": "SELECT * FROM snowflake.db.users WHERE user_id = 12", "timestamp": 1609459211, "user": "user_12", "operation_type": "SELECT"}
{"query": "SELECT * FROM snowflake.db.users WHERE user_id = 13", "timestamp": 1609459212, "user": "user_13", "operation_type": "SELECT"}
{"query": "SELECT * FROM snowflake.db.users WHERE user_id = 14", "timestamp": 1609459213, "user": "user_14", "operation_type": "SELECT"}
{"query": "SELECT * FROM snowflake.db.users WHERE user_id = 15", "timestamp": 1609459214, "user": "user_15", "operation_type": "SELECT"}

