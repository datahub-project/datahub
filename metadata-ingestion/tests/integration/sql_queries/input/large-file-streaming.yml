source:
  type: sql-queries
  config:
    query_file: ./input/large-file-streaming.jsonl
    platform: snowflake
    enable_streaming: true
    streaming_batch_size: 100
    use_schema_resolver: false

sink:
  type: file
  config:
    filename: ./output.json

datahub_api:
  server: http://localhost:8080
