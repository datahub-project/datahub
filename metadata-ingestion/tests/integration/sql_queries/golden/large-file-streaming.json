[
{
    "entityType": "dataset",
    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD)",
    "changeType": "UPSERT",
    "aspectName": "datasetUsageStatistics",
    "aspect": {
        "json": {
            "timestampMillis": 1609459200000,
            "eventGranularity": {
                "unit": "DAY",
                "multiple": 1
            },
            "partitionSpec": {
                "partition": "FULL_TABLE_SNAPSHOT",
                "type": "FULL_TABLE"
            },
            "uniqueUserCount": 15,
            "totalSqlQueries": 15,
            "topSqlQueries": [
                "SELECT * FROM snowflake.db.users WHERE user_id = 1",
                "SELECT * FROM snowflake.db.users WHERE user_id = 2",
                "SELECT * FROM snowflake.db.users WHERE user_id = 3",
                "SELECT * FROM snowflake.db.users WHERE user_id = 4",
                "SELECT * FROM snowflake.db.users WHERE user_id = 5",
                "SELECT * FROM snowflake.db.users WHERE user_id = 6",
                "SELECT * FROM snowflake.db.users WHERE user_id = 7",
                "SELECT * FROM snowflake.db.users WHERE user_id = 8",
                "SELECT * FROM snowflake.db.users WHERE user_id = 9",
                "SELECT * FROM snowflake.db.users WHERE user_id = 10"
            ],
            "userCounts": [
                {
                    "user": "urn:li:corpuser:user_1",
                    "count": 1
                },
                {
                    "user": "urn:li:corpuser:user_2",
                    "count": 1
                },
                {
                    "user": "urn:li:corpuser:user_3",
                    "count": 1
                },
                {
                    "user": "urn:li:corpuser:user_4",
                    "count": 1
                },
                {
                    "user": "urn:li:corpuser:user_5",
                    "count": 1
                },
                {
                    "user": "urn:li:corpuser:user_6",
                    "count": 1
                },
                {
                    "user": "urn:li:corpuser:user_7",
                    "count": 1
                },
                {
                    "user": "urn:li:corpuser:user_8",
                    "count": 1
                },
                {
                    "user": "urn:li:corpuser:user_9",
                    "count": 1
                },
                {
                    "user": "urn:li:corpuser:user_10",
                    "count": 1
                },
                {
                    "user": "urn:li:corpuser:user_11",
                    "count": 1
                },
                {
                    "user": "urn:li:corpuser:user_12",
                    "count": 1
                },
                {
                    "user": "urn:li:corpuser:user_13",
                    "count": 1
                },
                {
                    "user": "urn:li:corpuser:user_14",
                    "count": 1
                },
                {
                    "user": "urn:li:corpuser:user_15",
                    "count": 1
                }
            ],
            "fieldCounts": []
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062681,
        "runId": "sql-queries-2025_09_14-21_01_02-pwvcpj",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:4d299661aff121d683751de129f3a38159f74f2634315c0ae954ee85c4526b9b",
    "changeType": "UPSERT",
    "aspectName": "queryProperties",
    "aspect": {
        "json": {
            "customProperties": {},
            "statement": {
                "value": "SELECT * FROM snowflake.db.users WHERE user_id = 15",
                "language": "SQL"
            },
            "source": "SYSTEM",
            "created": {
                "time": 1609459214000,
                "actor": "urn:li:corpuser:user_15"
            },
            "lastModified": {
                "time": 1609459214000,
                "actor": "urn:li:corpuser:user_15"
            }
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062683,
        "runId": "sql-queries-2025_09_14-21_01_02-pwvcpj",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:4d299661aff121d683751de129f3a38159f74f2634315c0ae954ee85c4526b9b",
    "changeType": "UPSERT",
    "aspectName": "querySubjects",
    "aspect": {
        "json": {
            "subjects": [
                {
                    "entity": "urn:li:dataset:(urn:li:dataPlatform:snowflake,snowflake.db.users,PROD)"
                }
            ]
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062683,
        "runId": "sql-queries-2025_09_14-21_01_02-pwvcpj",
        "lastRunId": "no-run-id-provided"
    }
},
{
    "entityType": "query",
    "entityUrn": "urn:li:query:4d299661aff121d683751de129f3a38159f74f2634315c0ae954ee85c4526b9b",
    "changeType": "UPSERT",
    "aspectName": "dataPlatformInstance",
    "aspect": {
        "json": {
            "platform": "urn:li:dataPlatform:snowflake"
        }
    },
    "systemMetadata": {
        "lastObserved": 1757898062683,
        "runId": "sql-queries-2025_09_14-21_01_02-pwvcpj",
        "lastRunId": "no-run-id-provided"
    }
}
]